vllm/model_executor/models/molmo.py

```diff
@@ -242,7 +223,12 @@ def forward(self,
         if self._use_flash_attn:
             from flash_attn import flash_attn_func
             output = flash_attn_func(xq, xk, xv, dropout_p=0.0, causal=False)
-        else:
+        elif self._use_sdpa:
```

Comment:
Imo the `self._use_*` flags are redundant. Why not check against `selected_backend` directly?